view/frontend/web/js/view/payment/method-renderer/adyen-hpp-method.js

```diff
@@ -750,7 +747,18 @@ define(
                 if (paymentMethod.methodIdentifier.includes('amazonpay')) {
                     configuration.productType = 'PayAndShip';
                     configuration.checkoutMode = 'ProcessOrder';
-                    configuration.returnUrl = location.href;
+                    var url = new URL(location.href);
+                    if (url.searchParams.has('amazonCheckoutSessionId')) {
+                        url.searchParams.delete('amazonCheckoutSessionId');
+                    }
```

Comment:
We can omit the `has` check and always `delete`. If the parameter does not exist, `delete` accomplished it's job the easy way. :) 
```suggestion
                    url.searchParams.delete('amazonCheckoutSessionId');
```